frontends/web/src/api/backend.ts

```diff
@@ -31,6 +31,8 @@ export interface ISuccess {
     errorCode?: string;
 }
 
+export type TDecodedMessage = {success: boolean, decodedMessage: string, errorMessage: string}
```

Comment:
again here, why not `{ success: true; decodedMessage: string } | { success: false; errorMessage: string }`. typing more explicitly prevents one from accessing the wrong fields.